### PR TITLE
Regenerate website 2022/09/23

### PR DESCRIPTION
**Pull request by ras0219-msft**
This PR includes generated content; please review the following:
1. All changes to `scripts` (`scripts/rebuild.sh`, `scripts/validateLinks.js`)
2. One randomly selected html page for core navigation issues

Additionally, this fixes a regression from https://github.com/vcpkg/vcpkg.github.io/pull/81 which embeds nondeterminism (a current datetime stamp) into the output.json. When the CI check regenerates, this is different each time, so the check will always fail. This was not used on the website yet, so I've replaced it with the vcpkg commit id which could serve a similar purpose while being deterministic.

### CODE REVIEW

**Review by trustdaprocess93 — APPROVED** [review]

File: scripts/validateLinks.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ async function get_pages_recursive(docs_set, path) {
  * @returns {{links: [string,string][], fragments: {string}}}
```
Comment (BillyONeal) [inline scripts/validateLinks.js:35]: Missing other

Comment (BillyONeal) [reply scripts/validateLinks.js:35]: Also, should it be 'incorrect' instead?

File: scripts/validateLinks.js [hunk 2]
```diff
@@ -72,12 +72,15 @@ async function load_page_info(page, relative_path) {
     for (const match of content.matchAll(/ name="([^"]*)"/g)) {
         ret.fragments[match[1]] = true;
     }
+    for (const match of content.matchAll(/.{0,30}\]\[.{0,30}/g)) {
```
Comment (BillyONeal) [inline scripts/validateLinks.js:75]: Why the 0,30 constraint?

**Review by BillyONeal — APPROVED** [review]

File: scripts/validateLinks.js (hunk 2 above)
Comment (vicroms) [reply scripts/validateLinks.js:75]: prevent long text links? it is not obvious to me what this is doing either

Comment (ras0219-msft) [reply scripts/validateLinks.js:75]: It's just capturing context to display in the error message, so I wanted to limit it to a reasonable size.

**Review by vicroms — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**